### PR TITLE
[RFC] subsys: storage: flash_map: add API for check flash driver support

### PR DESCRIPTION
**Pull request by nvlsianpu**
Some applications might want to check whether flash_areas binds to
any flash drive in the system. It might be better to do that while
sanity check at application start-up then while regular run process.
Example of such application is the mcuboot.

This patch introduce such API for checking whether device bindings
were resolved properly during system startup.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

**Comment by nvlsianpu** [comment]
@d3zd3z  @utzig Using this API I want to replace hack sanity code here - https://github.com/runtimeco/mcuboot/blob/master/boot/zephyr/main.c#L104
This will allow to relay on flash_map abstraction only - so it was the last think which block external driver support for mcuboot@zephyr

**Comment by utzig** [comment]
LGTM

**Comment by nvlsianpu** [comment]
@lemrey In mcuboot we use static flash_map for now. It might changes in future. So I think it is more flexible to have this API.

**Review by lemrey — APPROVED** [review]
Looks good

**Review by carlescufi — APPROVED** [review]

File: include/flash_map.h [hunk 1]
```diff
@@ -77,6 +77,16 @@ u8_t flash_area_align(const struct flash_area *fa);
...
 
+/**
+ * Check whether given flash area has supporting flash driver
+ * in the system.
+ *
+ * @param fa Flash area.
+ *
+ * @return 0 On success. -ENODEV if no driver match.
+ */
+int flash_area_check_driver(const struct flash_area *fa);
```
Comment (carlescufi) [inline include/flash_map.h]: rename to `flash_area_has_driver` and return 1 on success

**Review by carlescufi — CHANGES_REQUESTED** [review]

**Comment by nvlsianpu** [comment]
@carlescufi done

**Review by carlescufi — APPROVED** [review]

**Pull request merged**